include/nnvm/tuple.h

```diff
@@ -83,6 +83,32 @@ class Tuple {
     this->SetDim(end - begin);
     std::copy(begin, end, this->begin());
   }
+  /*!
+   * \brief Assign content to tuple from iterator.
+   * \param begin the beginning of iterator
+   * \param end the end of the iterator
+   * \param d_first the beginning of the destination range
+   * \tparam RandomAccessIterator iterator type
+   */
+  template<typename RandomAccessIterator>
+  inline void assign(RandomAccessIterator begin,
+                     RandomAccessIterator end,
+                     ValueType* d_first) {
```

Comment:
This helper is bit abnormal, let us not include it in tuple as member, but put it as helper function somewhere.